DependencyInjection/Configuration.php

```diff
@@ -65,41 +64,27 @@ public function getConfigTreeBuilder()
                     ->end()
                 ->end()
                 ->arrayNode('document_tree')
-                    ->useAttributeAsKey('class')
-                    ->prototype('array')
-                        ->fixXmlConfig('valid_child', 'valid_children')
-                        ->children()
-                            ->arrayNode('valid_children')
-                                ->prototype('scalar')->end()
-                                ->info('class names of valid children, manage tree operations for them and hide other children')
-                            ->end()
-                            ->scalarNode('image')
-                                ->defaultValue('')
-                            ->end()
-                        ->end()
-                    ->end()
-                 ->end()
-
-                 ->scalarNode('document_tree_repository')->defaultValue('default')->end()
-
-                ->arrayNode('document_tree_defaults')
-                    ->prototype('scalar')->end()
-                ->end()
-
-                ->arrayNode('document_tree_options')
-                    ->addDefaultsIfNotSet()
+                    ->canBeEnabled()
                     ->children()
-                        ->integerNode('depth')
-                            ->defaultValue(1)
-                            ->info('Depth to which to fetch tree children when rendering the initial tree')
+                        ->scalarNode('repository_name')
+                            ->defaultValue('default')
+                            ->info('The repository name the resource API connects to.')
+                        ->end()
+                        ->arrayNode('routing_defaults')
+                            ->prototype('scalar')->end()
                         ->end()
-                        ->scalarNode('precise_children')
-                            ->defaultTrue()
-                            ->info('Exact check if document has children. For large trees, set to false for better performance, but user might needs to click on expand to see there are no children.')
+                        ->scalarNode('sortable_by')
+                            ->defaultValue('position')
+                            ->info('Defines the document property siblings are sorted.')
                         ->end()
-                        ->booleanNode('confirm_move')
-                            ->defaultTrue()
-                            ->info('Whether moving a node in the tree asks for confirmation.')
+                        ->arrayNode('move')
+                            ->canBeEnabled()
+                            ->children()
+                                ->booleanNode('reorder')
+                                    ->defaultTrue()
+                                    ->info('Ordering after a move action is enabled, causes new requests.')
```

Comment:
lets leave the explanations for the documentation and keep the info nodes minimal. "Whether to enable reordering sibling documents."

btw, i am not 100% sure this should be optional. i feel like if move is enabled, reordering should implicitly be allowed. what is the reasoning to make it optional?